### PR TITLE
Cleanup `dataMap` placement and imports

### PR DESCRIPTION
**Pull request by mrpiotr-dev**
### Context
As I mentioned in #8704, one of our clients uses Liferay's portlets. Their bundler uses a custom AMD loader that resolves `../dataMap` reference to `./src/dataMap/index`. As the result, `Core.loadData()` tries to create a new `DataMap` which is `undefined`.

This PR cleanups directories structure and adjusts paths to eliminate potential problems with circular references.

### How has this been tested?
I was able to reproduce the problem and test the solution only in the client's testing environment.
I manually tested the final package against possible regressions in ESM, CJS, UMD using webpack/rollup/tsc/babel.

// TODO: @handsontable/devs - It would be nice to have #7474 to automate tests for distribution files 🙃

### Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)

### Related issue(s):
1. #8704 

### Affected project(s):
- [x] `handsontable`

### CODE REVIEW

File: src/dataMap/metaManager/mods/__tests__/dynamicCellMeta.unit.js [hunk 1]
```diff
@@ -1,9 +1,9 @@
-import { DynamicCellMetaMod } from 'handsontable/dataMap/metaManager/mods/dynamicCellMeta';
-import MetaManager from 'handsontable/dataMap/metaManager';
-import Handsontable from 'handsontable';
-import Hooks from 'handsontable/pluginHooks';
+import { DynamicCellMetaMod } from '../dynamicCellMeta';
+import MetaManager from '../../';
+import Handsontable from '../../../../';
+import Hooks from '../../../../pluginHooks';
```
Comment (budnix) [inline src/dataMap/metaManager/mods/__tests__/dynamicCellMeta.unit.js:4]:
```suggestion
import MetaManager from '../../';
import Handsontable from 'handsontable';
import Hooks from '../../../../pluginHooks';
```

File: src/dataMap/metaManager/mods/__tests__/dynamicCellMeta.unit.js [hunk 2]
```diff
@@ -1,9 +1,9 @@
...
 
-jest.mock('handsontable');
+jest.mock('../../../../');
```
Comment (budnix) [inline src/dataMap/metaManager/mods/__tests__/dynamicCellMeta.unit.js:6]:
```suggestion
jest.mock('handsontable');
```

Comment (budnix) [reply src/dataMap/metaManager/mods/__tests__/dynamicCellMeta.unit.js:6]: Can we revert the 3 and 6 lines? I think that with a relative path, the mock loses its readability.

**Review by budnix — CHANGES_REQUESTED** [review]

File: src/dataMap/metaManager/mods/__tests__/dynamicCellMeta.unit.js (hunk 2 above)
Comment (mrpiotr-dev) [reply src/dataMap/metaManager/mods/__tests__/dynamicCellMeta.unit.js:6]: Of course! 🙂

**Review by budnix — APPROVED** [review]

**Review by budnix — APPROVED** [review]

**Pull request merged**